detectable/src/main/java/com/blackduck/integration/detectable/detectables/opam/buildexe/OpamBuildExtractor.java

```diff
@@ -0,0 +1,137 @@
+
+                  addProjectInformation(treeBuilder, result);
+                  return treeBuilder.build();
+              }
+          }
+
+          OpamFileParser opamFileParser = new OpamFileParser(); // initialize opam file parser
+
+          List<OpamParsedResult> opamParsedResults = new ArrayList<>();
+
+          for (File opamFile : opamFiles) {
+              opamParsedResults.add(opamFileParser.parse(opamFile)); // for all the opam files, parse direct dependencies found.
+          }
+
+          List<CodeLocation> codeLocations = new ArrayList<>();
+
+          for (OpamParsedResult opamParsedResult : opamParsedResults) {
+              codeLocations.add(opamGraphTransformer.transform(opamExe, opamParsedResult));
+          }
+
+          Extraction.Builder showBuilder = new Extraction.Builder().success(codeLocations);
+          addProjectInformation(showBuilder, opamParsedResults);
+
+          return showBuilder.build();
+      } catch (Exception e) {
+          return new Builder().exception(e).build();
+      }
+
+    }
+
+    private String getOpamVersion(ExecutableTarget opamExe) throws ExecutableRunnerException {
+        List<String> arguments = new ArrayList<>();
+        arguments.add("--version");
+
+        ExecutableOutput executableOutput = executableRunner.execute(ExecutableUtils.createFromTarget(sourceDirectory, opamExe, arguments));
+
+        if(executableOutput.getReturnCode() == 0) {
+            return executableOutput.getStandardOutputAsList().get(0);
+        } else {
+            logger.warn("Detect was not able to find opam version correctly, finding dependencies with opam show"); // if version command was not successful continue with opam show approach
+            return "";
+        }
+    }
+
+    public List<String> runOpamTree(ExecutableTarget opamExe) throws ExecutableRunnerException {
+        List<String> arguments = new ArrayList<>();
+        arguments.add("tree");
+        arguments.add(".");
+        arguments.add("--with-dev");
+        arguments.add("--with-test");
+        arguments.add("--with-doc");
+        arguments.add("--recursive");
+
+        ExecutableOutput executableOutput = executableRunner.execute(ExecutableUtils.createFromTarget(sourceDirectory, opamExe, arguments));
+
+        if(executableOutput.getReturnCode() == 0) {
+            return executableOutput.getStandardOutputAsList();
+        } else {
+            logger.warn("Detect was not able to find dependency tree correctly, finding dependencies with opam show"); // if tree command fails due to some reason, then we return empty list to continue with show approach
+            return Collections.emptyList();
+        }
+    }
+
+    private boolean decideOpamVersion(String opamVersion) {
+        if (!opamVersion.isEmpty()) {
+            String[] versionParts = opamVersion.split("\\.");
+
+            // check if version is greater than 2.2.0
```

Comment:
Not a problem as of now since it looks like there isn't Opan 3, but maybe worth considering that this method will return false for 3.0, 3.1, and then true again for 3.2.